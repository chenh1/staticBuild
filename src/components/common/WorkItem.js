import React, {PropTypes} from 'react';
import { Link, IndexLink } from 'react-router';

const WorkItem = ({path, uniqueClass, imageSrc, details, description}) => {
    return(
        <Link to={path} className="{uniqueClass} portfolio-item page-link">
            <div className="image">
                <img src={imageSrc} alt={details}></img>
            </div>
            <div className="info">
                <h3 className="header-three">{details}</h3>
                <p className="paragraph-one">
                    {description}
                </p>
            </div>
        </Link>
    );
}

export default WorkItem;