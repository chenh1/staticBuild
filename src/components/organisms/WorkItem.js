import React from 'react';
import PortfolioImage from '../molecules/PortfolioImage';
import PortfolioInfo from '../molecules/PortfolioInfo';

const WorkItem = ({data}) => {
    return (
        <div className="portfolio-item page-link">
            <PortfolioImage imageSrc={data.imageSrc} alt={data.alt}/>
            <PortfolioInfo subHeaderText={data.subHeader} paragraphText={data.description}/>
        </div>
    );
};

export default WorkItem;