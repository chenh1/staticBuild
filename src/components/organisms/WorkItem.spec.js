import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import WorkItem from './WorkItem';

test('renders portfolio item with image and portfolio info components', () => {
    const tree = renderer.create(<WorkItem
        path="/mockUrl"
        customClass="mock-class"
        imageSrc="/mockSrc"
        alt="Mock alt tag"
        subHeader="Mock Portfolio Item"
        description={["Some mock description"]} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});