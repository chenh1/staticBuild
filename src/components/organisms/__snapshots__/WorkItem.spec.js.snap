// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders portfolio item with image and portfolio info components 1`] = `
<a
  className="mock-class"
  onClick={[Function]}
  style={Object {}}
>
  <div
    className="image"
  >
    <img
      alt="Mock alt tag"
      src="/mockSrc"
    />
    
  </div>
  <div
    className="info"
  >
    <h3
      className="header-three"
    >
      Mock Portfolio Item
    </h3>
    <p
      className="paragraph-one"
    >
      Some mock description
    </p>
  </div>
</a>
`;
